platforms/jvm/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/AbstractJUnitTestClassProcessor.java

```diff
@@ -17,31 +17,37 @@
 package org.gradle.api.internal.tasks.testing.junit;
 
 import org.gradle.api.Action;
-import org.gradle.api.internal.tasks.testing.TestClassProcessor;
+import org.gradle.api.internal.tasks.testing.RequiresTestFrameworkTestClassProcessor;
 import org.gradle.api.internal.tasks.testing.TestClassRunInfo;
 import org.gradle.api.internal.tasks.testing.TestResultProcessor;
 import org.gradle.internal.actor.Actor;
 import org.gradle.internal.actor.ActorFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class AbstractJUnitTestClassProcessor implements TestClassProcessor {
+public abstract class AbstractJUnitTestClassProcessor implements RequiresTestFrameworkTestClassProcessor {
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractJUnitTestClassProcessor.class);
 
     private final ActorFactory actorFactory;
     private Actor resultProcessorActor;
     private Action<String> executor;
 
+    protected boolean startedProcessing;
+
     public AbstractJUnitTestClassProcessor(ActorFactory actorFactory) {
         this.actorFactory = actorFactory;
     }
 
     @Override
     public void startProcessing(TestResultProcessor resultProcessor) {
+        assertTestFrameworkAvailable();
```

Comment:
We should get rid of either this call or the call to assertTestFrameworkAvailable in createTestExecutor